modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/MoveModal.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {TreeView as ClayTreeView} from '@clayui/core';
+import ClayIcon from '@clayui/icon';
+import classnames from 'classnames';
+import {fetch, objectToFormData, openToast} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useMemo, useState} from 'react';
+
+import normalizeDropdownItems from '../utils/normalizeDropdownItems';
+import SearchField from './SearchField';
+
+const ITEM_TYPES_SYMBOL = {
+	article: 'document-text',
+	folder: 'folder',
+};
+
+const ITEM_TYPES = {
+	article: 'article',
+	folder: 'folder',
+};
+
+const showSuccessMessage = (portletNamespace) => {
+	const openToastSuccessProps = {
+		message: Liferay.Language.get('your-request-completed-successfully'),
+		type: 'success',
+	};
+
+	const reloadButtonLabel = Liferay.Language.get('reload');
+	const reloadButtonClassName = 'knowledge-base-reload-button';
+
+	openToastSuccessProps.message =
+		openToastSuccessProps.message +
+		`<div class="alert-footer">
+				<div class="btn-group" role="group">
+					<button class="btn btn-sm btn-primary alert-btn ${reloadButtonClassName}">${reloadButtonLabel}</button>
+				</div>
+		</div>`;
+
+	openToastSuccessProps.onClick = ({event, onClose: closeToast}) => {
+		if (event.target.classList.contains(reloadButtonClassName)) {
+			Liferay.Portlet.refresh(`#p_p_id${portletNamespace}`);
+			closeToast();
+		}
+	};
+
+	openToast(openToastSuccessProps);
+};
+
+const normalizeItems = (items) => {
+	if (items) {
+		return items.map((item) => {
+			return {
+				...item,
+				actions: normalizeDropdownItems(item.actions),
+				children: normalizeItems(item.children),
+			};
+		});
+	}
+};
+
+const getSearchItems = (items) => {
+	return items.reduce(function reducer(acc, item) {
+		acc.push({
+			href: item.href,
+			id: item.id,
+			name: item.name,
+			type: item.type,
+		});
+
+		if (item.children) {
+			item.children.reduce(reducer, acc);
+		}
+
+		return acc;
+	}, []);
+};
```

Comment:
These 3 functions are the same than in NavigationPanel. We can move them to /utils folder, where we have the `normalizeDropdownItems.js`
I would rename this file to `normalizeItems.js` and include here the function `normalizeItems`. And create new files `getSearchItems.js` and  `showSuccessMessage.js`.